core/transaction.go

```diff
@@ -409,6 +422,30 @@ func transactionCommitment(transactions []Transaction) (*felt.Felt, error) {
 	})
 }
 
+// blockVersionPost011 determines if the blocks protocolVersion is >=0.11.0.
+func blockVersionPost011(protocolVersion string) (bool, error) {
+
+	versionConstraint, _ := semver.NewConstraint(">=0.11.0")
+	var blockVersion *semver.Version
+	var err error
+
+	if protocolVersion == "" {
+		blockVersion, _ = semver.NewVersion("0.0.0")
+	} else {
+		sep := "."
+		digits := strings.Split(protocolVersion, sep)
+		// pad with 3 zeros in case version has less than 3 digits
+		digits = append(digits, []string{"0", "0", "0"}...)
+
+		// get first 3 digits only
+		blockVersion, err = semver.NewVersion(strings.Join(digits[:3], sep))
+		if err != nil {
+			return false, err
+		}
```

Comment:
instead of duplicating this piece here, you should make it a separate function, something like`parseBlockVersion` that takes the `string` and returns a `semver.Version`